src/main.ts

```diff
@@ -152,26 +127,27 @@ export default class VaultChat extends Plugin {
 			return []
 		}
 		const nearestVectors = this.vectorStore.getNearestVectors(embeddingResponse, 3, this.settings.relevanceThreshold)
-		const searchResults = Array.from(nearestVectors.keys())
-		const hydratedResults = []
-		for (const searchResult of searchResults) {
-			const abstractFile = this.app.vault.getAbstractFileByPath(searchResult)
-			if (!(abstractFile instanceof TFile)) {
-				console.warn(`Unexpected file type in search results. File: ${abstractFile?.name} `)
-				continue
+		const results = await Promise.all(nearestVectors.map(async (nearest, i) => {
+			let name = nearest.path.split('/').last() || ''
+			let contents = nearest.chunk
+			if (nearest.chunk && nearest.chunk.length) {
+				name = name + i // todo
```

Comment:
Right now, ChatGPT is referring the files in the response sometimes and its referring to them as like Ruby.md0 and Ruby.md1 because of this. Can we just repeat the base name in the map? It's not a key so it wouldn't break anything as far as I can tell, just seemed wonky. But then ChatGPT would call them all "Ruby.md" as expected.

Reply:
Maybe we can just rename before sending to ChatGPT. 🤔